internal/infrastructure/host/proxy_infra_test.go

```diff
@@ -141,6 +151,47 @@ func TestExtractSemver(t *testing.T) {
 	}
 }
 
+// TestInfra_runEnvoy_OutputRedirection verifies that Envoy output goes to configured writers, not os.Stdout/Stderr.
+func TestInfra_runEnvoy_OutputRedirection(t *testing.T) {
+	tmpdir := t.TempDir()
+	// Ensures that all the required binaries are available.
+	err := func_e.Run(t.Context(), []string{"--version"}, api.HomeDir(tmpdir))
+	require.NoError(t, err)
+
+	// Create separate buffers for stdout and stderr
+	stdout := &bytes.Buffer{}
+	stderr := &bytes.Buffer{}
+
+	i := &Infra{
+		proxyContextMap: make(map[string]*proxyContext),
+		HomeDir:         tmpdir,
+		Logger:          logging.DefaultLogger(stdout, egv1a1.LogLevelInfo),
+		Stdout:          stdout,
+		Stderr:          stderr,
+	}
+
+	// Run envoy with an invalid config to force it to write to stderr and exit quickly
+	args := []string{
+		"--config-yaml",
+		"invalid: yaml: syntax",
+	}
+
+	i.runEnvoy(t.Context(), "", "test", args)
+	require.Len(t, i.proxyContextMap, 1)
+
+	// Wait a bit for envoy to fail
+	require.Eventually(t, func() bool {
+		return stderr.Len() > 0 || stdout.Len() > 0
```

Comment:
Looks like there is a r/w race here... Worth wrapping the buffers into a sync struct?

Reply:
good catch. I imported a utility from aigw to fix that, and we can re-use the same elsewhere